src/config/parser.rs

```diff
@@ -51,19 +54,56 @@ impl Config {
     /// A function which parses the config.lua file and puts all the parsed options in the newly
     /// constructed Config struct and returns it.
     ///
+    /// # Arguments
+    ///
+    /// * `logging_initialized` - It takes a boolean which ensures that the logging doesn't get
+    /// initialized twice.
+    ///
     /// # Error
     ///
     /// Returns a lua parse error if parsing of the config.lua file fails or has a syntax error
     /// or io error if the config.lua file doesn't exists otherwise it returns a newly constructed
     /// Config struct with all the parsed config options from the parsed config file.
-    pub fn parse() -> Result<Self, Box<dyn std::error::Error>> {
+    pub fn parse(logging_initialized: bool) -> Result<Self, Box<dyn std::error::Error>> {
         Lua::new().context(|context| -> Result<Self, Box<dyn std::error::Error>> {
             let globals = context.globals();
 
             context
                 .load(&fs::read_to_string(Config::config_path()?)?)
                 .exec()?;
 
+            let parsed_threads: u8 = globals.get::<_, u8>("threads")?;
+
+            let debug: bool = globals.get::<_, bool>("debug")?;
+            let logging:bool= globals.get::<_, bool>("logging")?;
+
+            // Check whether logging has not been initialized before.
+            if logging_initialized {
+                // Initializing logging middleware with level set to default or info.
+                let mut log_level: LevelFilter = LevelFilter::Off;
+                if logging && debug == false {
+                    log_level = LevelFilter::Info;
+                } else if debug {
+                    log_level = LevelFilter::Trace;
+                };
+                env_logger::Builder::new().filter(None, log_level).init();
+            }
+
+            let threads: u8 = if parsed_threads == 0 {
+                    let total_num_of_threads:usize =  available_parallelism()?.get() /2;
+                if debug || logging {
```

Comment:
Do we have to check the `debug` and `logging` flag every time we want to add a log? It feels quite tedious. In my opinion, the `debug` and `logging` flags should only affect the logging level. We should be able to log in anywhere, considering only the log level we want to choose. @neon-mmd What do you think

Reply:
Yes, I think you are right, maybe we should make it log level based. But I think we should not do it in this PR because this PR will become huge as there are a lot of this checks in the program, and we will need to replace them all. :slightly_smiling_face: 

Also, I would like to have some suggestions on this. I think a `trace` log level is great, but after doing some research, I found that this log level should only be used when tracing third party libraries or code. So, I think `trace` the log level should be only when anyone builds the app with dev environment by allowing the `Cargo.toml` to set an `enviroment` variable as `dev`. This will then allow the program to read this variable and then turn on tracing. Otherwise, in production, there will only two logging level `debug` and `info`. What do you say ?? :slightly_smiling_face:

Here are some links on my research:
- https://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels
- https://sematext.com/blog/logging-levels/ 
- https://betterstack.com/community/guides/logging/log-levels-explained/

Reply:
> This will then allow the program to read this variable and then turn on tracing. Otherwise, in production, there will only two logging level `debug` and `info`. What do you say ?? 🙂

Yes, that makes sense. 

Additionally, thanks for providing the links, I have learned a lot from them